plugins/wacom-usb/fu-wac-device.c

```diff
@@ -726,6 +725,58 @@ fu_wac_device_add_modules(FuWacDevice *self, GError **error)
 		return FALSE;
 	}
 
+	/* validate versions of each submodule */
+	for (guint8 i = 0; i < buf[3]; i++) {
+		guint8 fw_type = buf[(i * 4) + 4] & ~0x80;
+		guint16 ver;
+
+		if (!fu_memread_uint16_safe(buf,
+					    sizeof(buf),
+					    (i * 4) + 5,
+					    &ver,
+					    G_BIG_ENDIAN,
+					    error))
+			return FALSE;
+
+		/* module is in transitional state or requires re-flashing */
+		if (ver == 0) {
+			g_set_error(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_FOUND,
```

Comment:
`FWUPD_ERROR_INVALID_DATA` perhaps? `not-found` is used when the device is removed and it might not do what you think it does.

Reply:
Yeah, I thought about it initially, but it is "Since: 2.0.0", and my  main target is 1.9.x

Reply:
Okay, use FWUPD_ERROR_INVALID_FILE for 1.9.x then -- I think FWUPD_ERROR_NOT_FOUND is going to cause subtle problems.

Reply:
or even FWUPD_ERROR_INTERNAL for 1.9.x

Reply:
I'm using error code to distinguish transitional error and FWUPD_ERROR_INTERNAL is used in other places, so FWUPD_ERROR_INVALID_FILE is good enough.
Hm... technically I can return the error with FWUPD_ERROR_INTERNAL, instead of just copying, so this code would live only for a short time inside the plugin.

Reply:
What do you think?

Reply:
FWUPD_ERROR_INTERNAL is fine for me